app/serializers/site_serializer.rb

```diff
@@ -251,12 +251,14 @@ def displayed_about_plugin_stat_groups
   end
 
   def anonymous_default_navigation_menu_tags
-    SiteSetting.default_navigation_menu_tags.split("|") - DiscourseTagging.hidden_tag_names(scope)
```

Comment:
This had the potential to ship down an empty array which can lead to subtle bugs on the client side since the client assumes that `anonymous_default_navigation_menu_tags` is a non-empty array.